src/library/scala/collection/ArrayOps.scala

```diff
@@ -1637,4 +1637,12 @@ final class ArrayOps[A](private val xs: Array[A]) extends AnyVal {
     *  @return `true` if this array has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = mutable.ArraySeq.make(xs).endsWith(that)
+
+  /** Ensure we are providing an array of the correct element type.
+   */
+  private def narrowly(ys: Array[_]): Array[A] =
+    if (implicitly[ClassTag[A]].runtimeClass.isAssignableFrom(ys.getClass.getComponentType))
+      ys.asInstanceOf[Array[A]]
+    else
+      Array[A](immutable.ArraySeq.unsafeWrapArray(ys).asInstanceOf[immutable.Seq[A]]: _*)
```

Comment:
if the argument is `mutable.ArraySeq[A]` I think we could just call `.toArray` here and be safer (no need for cast here).

Reply:
Thanks I forgot about `toArray`. You didn't hear the words I uttered aloud when asking myself how do I get an array. I wound up just doing `.toArray` instead of `.array`.